mkosi/__init__.py

```diff
@@ -1020,13 +1022,28 @@ def gen_kernel_images(state: MkosiState) -> Iterator[Tuple[str, Path]]:
         yield kver.name, kimg
 
 
-def initrd_path(state: MkosiState, kver: str) -> Path:
+def initrd_paths(state: MkosiState, kver: str) -> List[Path]:
+    if state.config.initrd:
+        return state.config.initrd
+
     # initrd file is versioned in Debian Bookworm
     initrd = state.root / boot_directory(state, kver) / f"initrd.img-{kver}"
     if not initrd.exists():
         initrd = state.root / boot_directory(state, kver) / "initrd"
 
-    return initrd
+    return [initrd]
+
+
+def kmods_initrd_path(state: MkosiState, kver: str) -> Optional[Path]:
+    if not state.config.initrd:
+        return None
+    return state.var_tmp() / f"initrd.{kver}.kmods.zst"
```

Comment:
The zst extension seems a bit too specific, which distro did you test this with?

Reply:
I actually control this extension. The "kmods" `initrd` is generated by `mkosi` for the kernel modules in new function [`generate_kmods_initrd`](https://github.com/systemd/mkosi/blob/47531782f9278eaf67d758148f44045827efc748/mkosi/__init__.py#L3547). This function generates a `cpio` archive from the `/usr/lib/modules` directory then compresses it with `zstd`.